src/main/java/org/generationcp/middleware/api/location/LocationServiceImpl.java

```diff
@@ -61,6 +64,27 @@ public long countFilteredLocations(final LocationSearchRequest locationSearchReq
 	@Override
 	public void deleteLocation(final Integer locationId) {
 		final Location location = this.daoFactory.getLocationDAO().getById(locationId);
+		List<Locdes> Blocks = null;
+		if (location.getLtype() == 415) { // Field
+			//Get the Blocks
+			Blocks = daoFactory.getLocDesDao().getLocdes(null, locationId.toString());
+			Blocks.forEach((locdes) -> {
+					this.daoFactory.getLocDesDao().makeTransient(locdes);
```

Comment:
@cuenyad sounds like we dont need to go n times to the database, we could have only 2 DELETE queries to perform these actions

Reply:
@clarysabel per field you can have more than one block.

Reply:
done.